### PR TITLE
feat: support multiple nodes in grafana dashboard

### LINKED ISSUE #2990: Support multiple nodes on the grafana dashboard
### Describe the feature

Currently the grafana dashboard is for one node which is fine, but ideally it would support multiple nodes via a drop down.

The graphs should show one value for each node, or the selected node in the drop down

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by leovct**
Closes #2990 

A new variable called `instance` has been added. It can either be equal to a node instance (e.g. `host.docker.internal:9001`), multiple node instances at the same time or it can be set to `All` (select all the nodes at the same time). 

Screenshot 2023-06-22 at 13 12 40

If the `instance` variable is set to a specific node instance, only the metrics related to this node will be shown.

Screenshot 2023-06-22 at 12 39 10

If you select all the instance  with `all`, you'll see that panels will be duplicated (e.g. `Database` will now become `Database (node X)` and `Database (node Y)`).

Screenshot 2023-06-22 at 12 39 02

On the screenshots, the first section is called Node, but I later changed the title to Overview.

### CODE REVIEW

**Comment by leovct** [comment]
Quick question, this section is empty in the dashboard, should I remove it?

Screenshot 2023-06-22 at 18 32 13

**Comment by shekhirin** [comment]
@Rjected @leovct I think it might be a useful context regarding JSON schema changes: https://github.com/paradigmxyz/reth/pull/1391#discussion_r1108147412

**Comment by leovct** [comment]
The merge didn't go well because there were too many changes between my branch et the current main branch. I'm going to quickly redo the changes and force push it from a clean branch.

**Comment by leovct** [comment]
> @Rjected @leovct I think it might be a useful context regarding JSON schema changes: [#1391 (comment)](https://github.com/paradigmxyz/reth/pull/1391#discussion_r1108147412)

Thanks, I exported the dashboard using Grafana. Since there's no tag in the docker-compose, I'm running the latest version and I think that's why there are so many changes on my PR.

https://github.com/paradigmxyz/reth/blob/main/etc/docker-monitoring.yml#L18

**Comment by shekhirin** [comment]
@leovct just to confirm, have you done it that way, checking the `Export for sharing externally` mark?

image

I'm on Grafana 10.0.1, which I believe is the latest version, and got a different resulting JSON on your PR: https://gist.github.com/shekhirin/1147069aabca4f5116f304d3bdc548b7

**Comment by leovct** [comment]
Just to explain what I did which was wrong, I was going to the Parameters > JSON Model and would copy-paste the code. I found the share button hopefully! 😄

Screenshot 2023-06-23 at 16 23 21

**Review by onbjerg — APPROVED** [review]
lgtm

**Review by shekhirin — CHANGES_REQUESTED** [review]
Something seems to be off with the JSON, looks like you've manually edited it:
```console
➜  reth (main) curl -s https://raw.githubusercontent.com/paradigmxyz/reth/afc3acb26c0b1c5b85a5e9adc03ffcbab6aa0daf/etc/grafana/dashboards/overview.json | jq
parse error: Objects must consist of key:value pairs at line 3832, column 13
```

I introduced a PR to prevent such errors in the future: https://github.com/paradigmxyz/reth/pull/3382, but let's fix that before merging.

**Comment by leovct** [comment]
> Something seems to be off with the JSON, looks like you've manually edited it:
> 
> ```
> ➜  reth (main) curl -s https://raw.githubusercontent.com/paradigmxyz/reth/afc3acb26c0b1c5b85a5e9adc03ffcbab6aa0daf/etc/grafana/dashboards/overview.json | jq
> parse error: Objects must consist of key:value pairs at line 3832, column 13
> ```
> 
> I introduced a PR to prevent such errors in the future: #3382, but let's fix that before merging.

Interesting... It may have to be related to the rebase I did earlier... Let me try to export the dashboard one more time!

File: etc/grafana/dashboards/overview.json
Comment (leovct) [reply etc/grafana/dashboards/overview.json:2636]: Ok, it's a bit painful to work on this issue because every time there's a commit that modifies `[etc/grafana/dashboards/overview.json](https://github.com/paradigmxyz/reth/pull/3336/files/64e902a073593964ccefe3e7621cb174428bc9da#diff-3694b6d47929cdc8535d9ab68c3ae13b86bfd87529024b4525484b413c351a2c)`, it breaks my PR and the rebase/merge never work unfortunately... I suggest opening a fresh PR with the changes and we try to merge it asap, what do you think?

Comment (shekhirin) [reply etc/grafana/dashboards/overview.json:2636]: I think continuing with this PR is fine. You can checkout to the latest main, and then re-apply your title, templating and `{instance=~\"$instance\"}` changes.

Comment (leovct) [reply etc/grafana/dashboards/overview.json]: @shekhirin Should be good now!

**Review by onbjerg — APPROVED** [review]
lgtm provided ci passes

**Review by shekhirin — APPROVED** [review]
LGTM, thank you! Love how clean the diff is now.

**Pull request merged**
